include/wil/registry_basic.h

```diff
@@ -1531,133 +1528,123 @@ inline HRESULT get_value_dword_nothrow(HKEY key, _In_opt_ PCWSTR value_name, _In
 
-    if (optionalvalue.type != optional_value_status::has_value)
-    {
-        RETURN_WIN32(ERROR_FILE_NOT_FOUND);
-    }
-
-    std::swap(data, optionalvalue.value);
+    ::std::swap(*data, registry_value.value);
     return S_OK;
 }
 CATCH_RETURN()
 
-inline HRESULT get_value_expanded_string_nothrow(HKEY key, _In_opt_ PCWSTR value_name, ::std::wstring& data) WI_NOEXCEPT
+inline HRESULT get_value_expanded_string_nothrow(HKEY key, _In_opt_ PCWSTR value_name, ::std::wstring* data) WI_NOEXCEPT
 {
     return get_value_expanded_string_nothrow(key, nullptr, value_name, data);
 }
 
-inline HRESULT get_value_multistring_wstring_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, ::std::vector<::std::wstring>* return_value) WI_NOEXCEPT try
+inline HRESULT get_value_multistring_wstring_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, _In_ ::std::vector<::std::wstring>* return_value) WI_NOEXCEPT try
 {
     return_value->clear();
 
     ::std::vector<BYTE> rawData;
-    RETURN_IF_FAILED(get_value_byte_vector_nothrow(key, subkey, value_name, REG_MULTI_SZ, rawData));
+    RETURN_IF_FAILED(get_value_byte_vector_nothrow(key, subkey, value_name, REG_MULTI_SZ, &rawData));
+
     if (!rawData.empty())
     {
         auto* const begin = reinterpret_cast<wchar_t*>(rawData.data());
         auto* const end = begin + rawData.size() / sizeof(wchar_t);
-        *return_value = ::wil::registry::details::get_wstring_vector_from_multisz(::std::vector<wchar_t>(begin, end));
+        *return_value = ::wil::reg::details::get_wstring_vector_from_multistring(::std::vector<wchar_t>(begin, end));
     }
+    else
+    {
+        printf("\n\trawData was empty!\n");
+    }
+
     return S_OK;
 }
 CATCH_RETURN()
 
-inline HRESULT get_value_multistring_wstring_nothrow(HKEY key, _In_opt_ PCWSTR value_name, ::std::vector<::std::wstring>* return_value) WI_NOEXCEPT try
+inline HRESULT get_value_multistring_wstring_nothrow(HKEY key, _In_opt_ PCWSTR value_name, _In_::std::vector<::std::wstring>* return_value) WI_NOEXCEPT
 {
     return get_value_multistring_wstring_nothrow(key, nullptr, value_name, return_value);
 }
 
-inline optional_value<::std::vector<::std::wstring>> get_value_multistring_wstring(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name)
+inline ::wil::reg::optional_value<::std::vector<::std::wstring>> try_get_value_multistring_wstring(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name)
 {
     ::std::vector<::std::wstring> local_value;
     const auto hr = get_value_multistring_wstring_nothrow(key, subkey, value_name, &local_value);
     if (SUCCEEDED(hr))
     {
-        optional_value<::std::vector<::std::wstring>> return_value;
-        return_value.type = optional_value_status::has_value;
-        std::swap(return_value.value, local_value);
+        ::wil::reg::optional_value<::std::vector<::std::wstring>> return_value;
+        return_value.value_assigned = true;

```

Comment:
`value_assigned` should automatically be set upon inserting/swapping a value into an `optional_value`; it shouldn't be set externally.